rc/src/lib.rs

```diff
@@ -0,0 +1,220 @@
+use std::ops;
+use std::cell::Cell;
+
+pub struct MyRc<T> {
+    data: *mut MyBox<T>,
+}
+
+pub struct MyWeak<T> {
+    data: *mut MyBox<T>,
+}
+
+struct MyBox<T> {
+    data: *mut T,
+    strong: Cell<usize>,
+    weak: Cell<usize>
+}
```

Comment:
This works, but introduces another layer of indirection. The ideal would be to have `data` be a `T`.